src/main/java/com/acme/edu/LoggerController.java

```diff
@@ -0,0 +1,56 @@
+package com.acme.edu;
+
+import com.acme.edu.command.Command;
+import com.acme.edu.saver.Saver;
+
+import java.util.Map;
+import java.util.HashMap;
+
+public class LoggerController {
+    private static LoggerController theOnlyLogger = new LoggerController();
```

Comment:
In real life you will not need to construct Singleton by yourself. Frameworks like Spring will make regular POJO singleton.